### PR TITLE
feat: add optional amqp 0.9.1 incoming messages support

### LINKED ISSUE #119: add optional amqp 0.9.1 incoming messages support

### PR DESCRIPTION
**Pull request by lepetitnuage31**
close #119

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -47,3 +47,4 @@ doc/plugins_guide.rst
 doc/CHANGELOG*.md
 doc/CHANGELOG*.rst
 doc/changelogs.rst
+.idea
```
Comment (thefab) [inline .gitignore]: ? what is it ?

File: config/config.ini [hunk 2]
```diff
@@ -137,6 +137,7 @@ logging=1
 install_switch=1
 install_ungzip=0
 install_mqtt_listener=0
+install_amqp_listener=1
```
Comment (thefab) [inline config/config.ini]:
```suggestion
install_amqp_listener=0
```

File: .gitignore (hunk 1 above)
Comment (lepetitnuage31) [reply .gitignore]: The .idea folder in the solution root contains IntelliJ’s project specific settings files.

File: config/config.ini (hunk 2 above)
Comment (lepetitnuage31) [reply config/config.ini]: change made

File: config/config.ini [hunk 3]
```diff
@@ -168,6 +172,33 @@ mqtt_listener_broker_port=1883
...
+####################
+#####   AMQP   #####
+####################
+# the hostname or IP address of the remote broker
+amqp_consumer_broker_hostname=localhost
+
+# the network port of the server host to connect to. Defaults to 5672.
+amqp_consumer_broker_port=5672
+
+# a string specifying the credential user.
```
Comment (thefab) [inline config/config.ini]:
```suggestion
# The username to use to connect to the broker
```

File: config/config.ini [hunk 4]
```diff
@@ -168,6 +172,33 @@ mqtt_listener_broker_port=1883
...
+amqp_consumer_credential_user=guest
+
+# a string specifying the credential password
```
Comment (thefab) [inline config/config.ini]:
```suggestion
# The password to use to connect to the broker (use \ to escape special chars)
```

File: config/config.ini [hunk 5]
```diff
@@ -168,6 +172,33 @@ mqtt_listener_broker_port=1883
...
+amqp_consumer_credential_password=\$guest
+
+# a string specifying the subscription exchange.
```
Comment (thefab) [inline config/config.ini]:
```suggestion
# The subscription exchange name
```

File: config/config.ini [hunk 6]
```diff
@@ -168,6 +172,33 @@ mqtt_listener_broker_port=1883
...
+amqp_consumer_subscription_exchange=test
+
+# a string specifying the subscription type exchange. (fanout or topic default is fanout)
```
Comment (thefab) [inline config/config.ini]:
```suggestion
# The subscription exchange type (fanout or topic)
```

File: config/config.ini [hunk 7]
```diff
@@ -168,6 +172,33 @@ mqtt_listener_broker_port=1883
...
+amqp_consumer_subscription_exchange_type=fanout
+
+# a string specifying the subscription queue. (mode fanout)
```
Comment (thefab) [inline config/config.ini]:
```suggestion
# The subscription queue (for "fanout" mode)
```

File: config/config.ini [hunk 8]
```diff
@@ -168,6 +172,33 @@ mqtt_listener_broker_port=1883
...
+amqp_consumer_subscription_queue=my_queue
+
+# strings specifying the subscription routing keys. (mode topic)
```
Comment (thefab) [inline config/config.ini]:
```suggestion
# The subscription routing key (wildcards allowed) (for "topic" mode)
```

File: config/config.ini [hunk 9]
```diff
@@ -168,6 +172,33 @@ mqtt_listener_broker_port=1883
 # a string specifying the subscription topic to subscribe to.Default to # (all)
 mqtt_listener_subscription_topic=#
 
+####################
+#####   AMQP   #####
+####################
+# the hostname or IP address of the remote broker
+amqp_consumer_broker_hostname=localhost
+
+# the network port of the server host to connect to. Defaults to 5672.
```
Comment (thefab) [inline config/config.ini]:
```suggestion
# the network port of the server host to connect to
```

**Pull request merged**